consensus/statehash.go

```diff
@@ -0,0 +1,96 @@
+package consensus
+
+import (
+	"sort"
+
+	"github.com/NebulousLabs/Sia/hash"
+)
+
+// Sorted UtxoSet returns all of the unspent transaction outputs sorted
+// according to the numerical value of their id.
+func (s *State) sortedUtxoSet() (sortedOutputs []Output) {
+	var unspentOutputStrings []string
+	for outputID := range s.unspentOutputs {
+		unspentOutputStrings = append(unspentOutputStrings, string(outputID[:]))
+	}
+	sort.Strings(unspentOutputStrings)
+
+	for _, utxoString := range unspentOutputStrings {
+		var outputID OutputID
+		copy(outputID[:], utxoString)
+		output, err := s.output(outputID)
+		if err != nil {
+			panic(err)
+		}
+		sortedOutputs = append(sortedOutputs, output)
+	}
+	return
+}
+
+// StateHash returns the markle root of the current state of consensus.
+func (s *State) stateHash() hash.Hash {
```

Comment:
is it a problem that state.stateHash returns a hash.Hash?

For now I would like to keep the name, but will keep thinking about it. Mostly, there are a lot of things that could be hashed. It should really probably be called consensusHash